dascore/utils/array.py

```
@@ -232,20 +290,24 @@ def _ensure_array_compatible(patch, other):
+        # Scalars broadcast against anything.
+        if not shape or patch.shape == shape:
             return patch
-        if (patch_ndims := patch.ndim) < (array_ndims := other.ndim):
+        if (patch_ndims := patch.ndim) < (array_ndims := len(shape)):
             msg = f"Cannot broadcast patch/array {patch_ndims=} {array_ndims=}"
             raise PatchBroadcastError(msg)
-        patch = patch.make_broadcastable_to(other.shape)
+        patch = patch.make_broadcastable_to(shape)
         return patch
 
     def _apply_op(array1, array2, operator, reversed=False):
```

Ruff (0.16.1)

[error] Function argument `reversed` is shadowing a Python builtin

(A002)